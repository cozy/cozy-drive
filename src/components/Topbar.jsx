import styles from '../styles/topbar'

import React from 'react'
import { connect } from 'react-redux'

import FilesToolbar from '../containers/FilesToolbar'
import TrashToolbar from '../containers/TrashToolbar'
import Breadcrumb from '../containers/Breadcrumb'

import { FILES_CONTEXT, TRASH_CONTEXT } from '../constants/config'

const TOOLBARS = {
  [FILES_CONTEXT]: FilesToolbar,
  [TRASH_CONTEXT]: TrashToolbar
}

const Topbar = ({ context }) => {
  const Toolbar = TOOLBARS[context]
  return (
    <div class={styles['fil-content-header']}>
      <Breadcrumb />
      <Toolbar />
    </div>
  )
}

const mapStateToProps = (state, ownProps) => ({
  context: state.context
})

export default connect(mapStateToProps)(Topbar)
