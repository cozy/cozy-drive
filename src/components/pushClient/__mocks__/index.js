export const track = jest.fn()

export const isLinux = jest.fn()
export const isAndroid = jest.fn()
export const isIOS = jest.fn()

export const DESKTOP_BANNER = 'desktop_banner'
export const NOVIEWER_DESKTOP_CTA = 'noviewer_desktop_cta'

export const isClientAlreadyInstalled = jest.fn().mockResolvedValue(false)
