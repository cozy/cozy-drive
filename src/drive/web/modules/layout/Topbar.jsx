/* global __TARGET__ */
import React from 'react'
import classNames from 'classnames'

import styles from 'drive/styles/topbar.styl'

const Topbar = ({ children, hideOnMobile = true }) => (
  <div
    className={classNames(styles['fil-topbar'], {
      [styles['mobile']]: __TARGET__ === 'mobile',
      [styles['hidden-mobile']]: hideOnMobile
    })}
  >
    {children}
  </div>
)

export default Topbar
