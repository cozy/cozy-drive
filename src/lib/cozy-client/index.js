export { default as CozyProvider } from './CozyProvider'
export { default as CozyClient } from './CozyClient'
export { default as cozyConnect } from './connect'
export { default as cozyMiddleware } from './middleware'
export {
  default as reducer,
  makeActionCreator,
  fetchCollection,
  refetchCollections,
  fetchDocument,
  fetchReferencedFiles,
  addReferencedFiles,
  removeReferencedFiles,
  applySelectorForAction,
  enhancePropsForActions,
  getCollection,
  getDocument,
  createDocument,
  updateDocument,
  deleteDocument,
  createFile,
  trashFile,
  CREATE_DOCUMENT
} from './reducer'
export {
  fetchSharings,
  fetchContacts,
  getSharingDetails,
  share,
  unshare,
  leave,
  shareByLink,
  revokeLink
} from './slices/sharings'
export {
  startSync,
  isFirstSync,
  isSynced,
  hasSyncStarted
} from './slices/synchronization'

export { downloadArchive, downloadFile } from './helpers'
