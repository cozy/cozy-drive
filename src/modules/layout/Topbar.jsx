import classNames from 'classnames'
import React from 'react'

import styles from 'styles/topbar.styl'

const Topbar = ({ children, hideOnMobile = true }) => (
  <div
    className={classNames('u-mb-1-half', styles['fil-topbar'], {
      [styles['hidden-mobile']]: hideOnMobile
    })}
  >
    {children}
  </div>
)

export default Topbar
