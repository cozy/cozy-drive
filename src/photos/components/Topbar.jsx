/* global cozy  */
import styles from '../styles/topbar'

import React, { Component } from 'react'
import { translate } from 'cozy-ui/react/I18n'
import { withBreakpoints } from 'cozy-ui/react'
import { withRouter } from 'react-router'
import PropTypes from 'prop-types'
import flow from 'lodash/flow'

const { BarCenter, BarRight, BarLeft } = cozy.bar

const KEYCODE_ENTER = 13
const KEYCODE_ESC = 27

class EditableAlbumName extends Component {
  constructor(props) {
    super(props)
    this.ignoreBlurEvent = false // we'll ignore blur events if they are triggered by pressing enter or escape, to prevent `onEdit` from firing twice
    this.inputElement = null
  }

  componentDidMount() {
    if (this.inputElement !== null) {
      this.inputElement.focus()
      this.inputElement.select()
    }
  }

  handleBlur = e => {
    if (!this.ignoreBlurEvent && this.props.onEdit)
      this.props.onEdit(
        e.target.value.trim() !== '' ? e.target.value : this.props.albumName
      )
  }

  handleKeyDown = e => {
    if (e.keyCode === KEYCODE_ENTER && this.props.onEdit) {
      this.ignoreBlurEvent = true
      this.props.onEdit(e.target.value)
    } else if (e.keyCode === KEYCODE_ESC && this.props.onEdit) {
      this.ignoreBlurEvent = true
      this.props.onEdit(this.props.albumName)
    }
  }

  render() {
    const { albumName } = this.props
    return (
      <input
        type="text"
        defaultValue={albumName}
        onKeyDown={this.handleKeyDown}
        onBlur={this.handleBlur}
        ref={elem => {
          this.inputElement = elem
        }}
      />
    )
  }
}

EditableAlbumName.propTypes = {
  albumName: PropTypes.string,
  onEdit: PropTypes.func.isRequired
}

const TopbarTitle = ({ children }) => (
  <h2 data-test-id="pho-content-title" className={styles['pho-content-title']}>
    {children}
  </h2>
)

const BackToAlbumsButton = ({ onClick }) => (
  <div
    role="button"
    data-test-id="pho-content-album-previous"
    className={styles['pho-content-album-previous']}
    onClick={onClick}
  />
)

BackToAlbumsButton.propTypes = {
  onClick: PropTypes.func.isRequired
}

class Topbar extends Component {
  componentDidMount() {
    const url = this.props.router.location.pathname
    this.parentUrl = url.substring(0, url.lastIndexOf('/'))
  }

  renderTitle() {
    const { t, viewName, albumName = '', onEdit, editing = false } = this.props
    const isAlbumContent = viewName === 'albumContent'

    if (!isAlbumContent) return t(`Nav.${viewName}`)
    else if (editing)
      return <EditableAlbumName albumName={albumName} onEdit={onEdit} />
    else return albumName
  }

  render() {
    const {
      children,
      viewName,
      breakpoints: { isMobile },
      router
    } = this.props
    const isAlbumContent = viewName === 'albumContent'
    const title = <TopbarTitle>{this.renderTitle()}</TopbarTitle>
    const responsiveTitle = isMobile ? <BarCenter>{title}</BarCenter> : title

    const responsiveMenu = isMobile ? <BarRight>{children}</BarRight> : children

    const backButton = (
      <BackToAlbumsButton onClick={() => router.push(this.parentUrl)} />
    )
    const responsiveBackButton = isMobile ? (
      <BarLeft>{backButton}</BarLeft>
    ) : (
      backButton
    )

    return (
      <div className={styles['pho-content-header']}>
        {isAlbumContent && responsiveBackButton}
        {responsiveTitle}
        {responsiveMenu}
      </div>
    )
  }
}

Topbar.propTypes = {
  viewName: PropTypes.string.isRequired,
  albumName: PropTypes.string,
  t: PropTypes.func.isRequired,
  editing: PropTypes.bool,
  onEdit: PropTypes.func,
  breakpoints: PropTypes.object.isRequired,
  router: PropTypes.object.isRequired,
  children: PropTypes.node
}

Topbar.defaultProps = {
  editing: false,
  onEdit: () => {}
}

export default flow(
  withRouter,
  withBreakpoints(),
  translate()
)(Topbar)
