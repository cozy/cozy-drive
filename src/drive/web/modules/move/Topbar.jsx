import React from 'react'
import PropTypes from 'prop-types'
import get from 'lodash/get'
import Topbar from 'drive/web/modules/layout/Topbar'
import { withBreakpoints } from 'cozy-ui/react'
import {
  Breadcrumb,
  PreviousButton,
  renamePathNames
} from 'drive/web/modules/navigation/Breadcrumb'
import getFolderPath from 'drive/web/modules/navigation/getFolderPath'

const buildBreadcrumbPath = (currentDir, t) =>
  renamePathNames(
    getFolderPath({
      ...currentDir,
      parent: get(currentDir, 'relationships.parent.data')
    }),
    '',
    t
  )

const MoveTopbar = (
  { navigateTo, currentDir, fetchStatus, breakpoints: { isMobile } },
  { t }
) => {
  const path =
    fetchStatus === 'loaded' ? buildBreadcrumbPath(currentDir, t) : []
  const showPreviousButton = path.length > 1 && isMobile
  return (
    <Topbar hideOnMobile={false}>
      {showPreviousButton && (
        <PreviousButton onClick={() => navigateTo(path[path.length - 2])} />
      )}
      <Breadcrumb
        path={path}
        onBreadcrumbClick={navigateTo}
        opening={false}
        inlined
      />
    </Topbar>
  )
}

MoveTopbar.propTypes = {
  navigateTo: PropTypes.func.isRequired,
  currentDir: PropTypes.object.isRequired,
  fetchStatus: PropTypes.string.isRequired,
  breakpoints: PropTypes.shape({
    isMobile: PropTypes.bool
  }).isRequired
}
MoveTopbar.contextTypes = {
  t: PropTypes.func.isRequired
}
export default withBreakpoints()(MoveTopbar)
