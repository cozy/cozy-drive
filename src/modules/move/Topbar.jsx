import PropTypes from 'prop-types'
import React from 'react'

import { useQuery } from 'cozy-client'
import Icon from 'cozy-ui/transpiled/react/Icon'
import IconButton from 'cozy-ui/transpiled/react/IconButton'
import FolderAddIcon from 'cozy-ui/transpiled/react/Icons/FolderAdd'
import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'

import BackButton from 'components/Button/BackButton'
import Topbar from 'modules/layout/Topbar'
import { getBreadcrumbPath } from 'modules/move/helpers'
import Breadcrumb from 'modules/navigation/Breadcrumb/Breadcrumb'
import { buildOnlyFolderQuery } from 'modules/queries'

const MoveTopbar = ({ navigateTo, folderId, showFolderCreation }) => {
  const { t } = useI18n()
  const { isMobile } = useBreakpoints()
  const folderQuery = buildOnlyFolderQuery(folderId)
  const { fetchStatus, data } = useQuery(
    folderQuery.definition,
    folderQuery.options
  )

  const path = fetchStatus === 'loaded' ? getBreadcrumbPath(t, data) : []
  const showPreviousButton = path.length > 1 && isMobile

  return (
    <Topbar hideOnMobile={false}>
      {showPreviousButton && (
        <BackButton onClick={() => navigateTo(path[path.length - 2])} t={t} />
      )}
      <Breadcrumb
        path={path}
        onBreadcrumbClick={navigateTo}
        opening={false}
        inlined
      />
      <IconButton onClick={showFolderCreation} aria-label={t('Move.addFolder')}>
        <Icon icon={FolderAddIcon} />
      </IconButton>
    </Topbar>
  )
}

MoveTopbar.propTypes = {
  navigateTo: PropTypes.func.isRequired,
  folderId: PropTypes.string.isRequired
}

export default MoveTopbar
