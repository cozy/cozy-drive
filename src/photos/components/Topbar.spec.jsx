import React from 'react'
import { mount } from 'enzyme'
import AppLike from '../../../test/components/AppLike'
import { mockedRouter } from '../../../test/__mocks__/mockedRouter'
import { Topbar } from './Topbar'
import { createMockClient } from 'cozy-client'
describe('Topbar', () => {
  let component
  const client = createMockClient({})

  const setup = isMobile => {
    const Child = props => <div {...props}> Child </div>
    component = mount(
      <AppLike client={client}>
        <Topbar
          breakpoints={{ isMobile: isMobile }}
          router={mockedRouter}
          viewName="albumContent"
          t={x => x}
          client={client}
        >
          <Child router={mockedRouter} />
        </Topbar>
      </AppLike>
    )
  }
  it('should render the topbar for desktop', () => {
    setup(false)
    expect(component.render()).toMatchSnapshot()
  })

  it('renders the topbar for mobile and the child have the expected router', () => {
    setup(true)
    expect(component.render()).toMatchSnapshot()
    expect(component.find('Child').prop('router')).toBe(mockedRouter)
  })
})
