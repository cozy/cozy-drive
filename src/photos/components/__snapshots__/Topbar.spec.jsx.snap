// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar renders the topbar for mobile and the child have the expected router 1`] = `
<div
  class="pho-content-header"
>
  <div
    router="[object Object]"
  >
     Child 
  </div>
</div>
`;

exports[`Topbar should render the topbar for desktop 1`] = `
<div
  class="pho-content-header"
>
  <div
    class="pho-content-album-previous"
    data-test-id="pho-content-album-previous"
    role="button"
  />
  <h2
    class="pho-content-title"
    data-test-id="pho-content-title"
  />
  <div
    router="[object Object]"
  >
     Child 
  </div>
</div>
`;
